### DISCUSSION TITLE
Puppet8 Open source(Puppet agent can't connect to master)

### DISCUSSION THREAD

**Discussion by rbmramesh** [puppetlabs/community#50] [Troubleshooting]
Today i have installed puppetserver8 on ubuntu22.04LTS machine. I have configured puppet.conf file in puppet server as like below[1] and i allowed 8140 in firewall. Next i have installed puppet agent8 in client system and configured like below[2] . Now i tried to sync with puppet master but i am getting following error [3].
[1]
```
root@ultra:~# cat /etc/puppetlabs/puppet/puppet.conf

[server]
vardir = /opt/puppetlabs/server/data/puppetserver
logdir = /var/log/puppetlabs/puppetserver
rundir = /var/run/puppetlabs/puppetserver
pidfile = /var/run/puppetlabs/puppetserver/puppetserver.pid
codedir = /etc/puppetlabs/code

dns_alt_names = ultra.bahwan.org

[main]
certname = ultra
server = ultra.bahwan.org
environment = production
runinterval = 30m
root@ultra:~#
```

[2] Client configuration
```
root@sync0:# cat /etc/puppetlabs/puppet/puppet.conf
server=ultra.bahwan.org
environment = production
runinterval = 30m
root@sync0:#
```

[3] Error message :
```
Error: Connection to https://ultra.bahwan.org:8140/puppet-ca/v1 failed, trying next route: Request to https://ultra.bahwan.org:8140/puppet-ca/v1 failed after 0.641 seconds: SSL_connect returned=1 errno=0 peeraddr=10.177.55.78:8140 state=error: unexpected eof while reading
Wrapped exception:
SSL_connect returned=1 errno=0 peeraddr=10.177.46.78:8140 state=error: unexpected eof while reading
Error: No more routes to ca
Error: No more routes to ca
```

I changed cat /etc/default/puppetserver congiuration like below
Modify this if you'd like to change the memory allocation, enable JMX, etc

JAVA_ARGS="-Xms4g -Xmx4g -Djruby.logger.class=com.puppetlabs.jruby_utils.jruby.Slf4jLogger"

I have tried but i am not getting solution. Can you please help on this?
Note: I updated both a machine time and allowed 8140 in iptables file.

**Comment by vladsamoylik** [comment]
Please try it:

/etc/puppetlabs/puppetserver/services.d/ca.cfg

```
# To enable the CA service, leave the following line uncommented
puppetlabs.services.ca.certificate-authority-service/certificate-authority-service
# To disable the CA service, comment out the above line and uncomment the line below
#puppetlabs.services.ca.certificate-authority-disabled-service/certificate-authority-disabled-service
puppetlabs.trapperkeeper.services.watcher.filesystem-watch-service/filesystem-watch-service
```

**Comment by rbmramesh** [comment]
Hi vladsamoylik,
          The file is already looks like as you mentioned.

# To enable the CA service, leave the following line uncommented
puppetlabs.services.ca.certificate-authority-service/certificate-authority-service
# To disable the CA service, comment out the above line and uncomment the line below
#puppetlabs.services.ca.certificate-authority-disabled-service/certificate-authority-disabled-service
puppetlabs.trapperkeeper.services.watcher.filesystem-watch-service/filesystem-watch-service

 I deleted CA file and regenerated and deleted ssl folder in puppet server and agent.  I am getting same error.

root@sync0:~# puppet agent -t  --http_debug
opening connection to ultra.bahwan.org:8140...
opened
starting SSL for ultra.bahwan.org:8140...
Conn close because of connect error SSL_connect returned=1 errno=0 peeraddr=10.177.55.78:8140 state=error: unexpected eof while reading
Error: Connection to https://ultra.bahwan.org:8140/puppet-ca/v1 failed, trying next route: Request to https://ultra.bahwan.org:8140/puppet-ca/v1 failed after 0.434 seconds: SSL_connect returned=1 errno=0 peeraddr=10.177.55.78:8140 state=error: unexpected eof while reading
Wrapped exception:
SSL_connect returned=1 errno=0 peeraddr=10.177.55.78:8140 state=error: unexpected eof while reading
Error: No more routes to ca
Error: No more routes to ca
